core/client-internal/src/main/java/alluxio/client/UfsUtils.java

```diff
@@ -42,177 +40,70 @@
+        throw e;
       }
+      LOG.warn("Mount point: " + mountPointUri + " has already been mounted! Assuming the mount "
+          + "is consistent and proceeding to load metadata.");
     }
-
-    Queue<AlluxioURI> ufsPathQueue = new LinkedList<AlluxioURI>();
-    if (excludePathPrefix.outList(ufsRootPath)) {
-      ufsPathQueue.add(ufsAddrRootPath);
-    }
-
-    while (!ufsPathQueue.isEmpty()) {
-      AlluxioURI ufsPath = ufsPathQueue.poll(); // this is the absolute path
-      LOG.debug("Loading: {}", ufsPath);
-      if (ufs.isFile(ufsPath.toString())) { // TODO(hy): Fix path matching issue.
-        AlluxioURI tfsPath = buildTFSPath(directoryName, ufsAddrRootPath, ufsPath);
-        LOG.debug("Loading ufs. fs path = {}.", tfsPath);
-        if (fs.exists(tfsPath)) {
-          LOG.debug("File {} already exists in Alluxio.", tfsPath);
-          continue;
-        } else {
-          fs.loadMetadata(tfsPath);
-        }
-        LOG.debug("Create alluxio file {} with checkpoint location {}", tfsPath, ufsPath);
-      } else { // ufsPath is a directory
-        LOG.debug("Loading ufs. ufs path is a directory.");
-        String[] files = ufs.list(ufsPath.toString()); // ufs.list() returns relative path
-        if (files != null) {
-          for (String filePath : files) {
-            if (filePath.isEmpty()) { // Prevent infinite loops
-              continue;
-            }
-            LOG.debug("Get: {}", filePath);
-            String aPath = PathUtils.concatPath(ufsPath, filePath);
-            String checkPath = aPath.substring(ufsAddrRootPath.toString().length());
-            if (checkPath.startsWith(AlluxioURI.SEPARATOR)) {
-              checkPath = checkPath.substring(AlluxioURI.SEPARATOR.length());
-            }
-            if (excludePathPrefix.inList(checkPath)) {
-              LOG.debug("excluded: {}", checkPath);
-            } else {
-              ufsPathQueue.add(new AlluxioURI(aPath));
-            }
-          }
-        }
-        // ufsPath is a directory, so only concat the tfsRoot with the relative path
-        AlluxioURI tfsPath = new AlluxioURI(PathUtils.concatPath(
-            alluxioPath, ufsPath.getPath().substring(ufsAddrRootPath.getPath().length())));
-        LOG.debug("Loading ufs. ufs path is a directory. tfsPath = {}.", tfsPath);
-        if (!fs.exists(tfsPath)) {
-          LOG.debug("Loading ufs. ufs path is a directory. make dir = {}.", tfsPath);
-          fs.loadMetadata(tfsPath);
-          // TODO(hy): Add the following.
-          // if (fs.mkdir(tfsPath)) {
-          // LOG.info("Created Alluxio folder {} with checkpoint location {}", tfsPath, ufsPath);
-          // } else {
-          // LOG.info("Failed to create alluxio folder: {}", tfsPath);
-          // }
+    // Get the list of files to load
+    List<String> files = Arrays.asList(ufs.listRecursive(mountDirectoryPath));
+    for (String file : files) {
+      LOG.info("Considering file: " + file);
```

Comment:
this might result in excessive logging ... it might make sense to change this to LOG.debug